__tests__/integration/utils/test-utils-integration.js

```diff
@@ -53,6 +54,18 @@ export class TestUtils {
         if (err) {
           return reject(err);
         }
+        superagent.parse['application/json'] = (res, callback) => {
+          let rawData = '';
+          res.on('data', chunk => { rawData += chunk; });
+          res.on('end', () => {
+            try {
+              res.text = rawData;
+              callback(null, bigIntUtils.JSONBigInt.parse(rawData));
+            } catch (parseErr) {
+              callback(parseErr);
+            }
+          });
+        };
```

Comment:
This is slightly different than the default implementation (https://github.com/ladjs/superagent/blob/master/src/node/parsers/json.js). Are there any reasons for that?